src/lib/io/sol-iio.c

```diff
@@ -1107,10 +1111,12 @@ channel_get_offset(struct sol_iio_channel *channel)
 
 #define GET_OFFSET(_name) \
     do { \
-        r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
-            device->device_id, _name); \
-        if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
-            goto end; \
+        if (_name) { \
```

Comment:
@ceolin This checking is used for the second GET_OFFSET call:
    channel_pure_name = channel_get_pure_name(channel);
    GET_OFFSET(channel_pure_name);

I found the below log during testing, since the channel_get_pure_name returns NULL
DBG: ./src/lib/io/sol-iio.c:165 craft_filename_path() available=4096, used=46, path='/sys/bus/iio/devices/iio:device1/(null)_offset'